src/interface_elements.jl

```diff
@@ -0,0 +1,456 @@
+    get_sides_and_base_indices(chere::AbstractCell, cthere::AbstractCell)
+    get_sides_and_base_indices(::Type{<:AbstractRefShape}, ::Type{<:AbstractRefShape})
+
+Return a tuple containing tuples of a symbol (:here or :there) and an integer.
+The index of the outer tuple represents the node index.
+In the inner tuple, the symbol represents the side the node is on 
+and the integer represents the nodes index in the base cell.
+"""
+get_sides_and_base_indices(c::InterfaceCell) = get_sides_and_base_indices(c.here, c.there)
+get_sides_and_base_indices(chere::AbstractCell, cthere::AbstractCell) = get_sides_and_base_indices(typeof(chere), typeof(cthere))
+
+get_sides_and_base_indices(::Type{Line}, ::Type{Line}) = ((:here,1), (:here,2), (:there,1), (:there,2))
+get_sides_and_base_indices(::Type{QuadraticLine}, ::Type{Line}) = ((:here,1), (:here,2), (:there,1), (:there,2), (:here,3))
+get_sides_and_base_indices(::Type{Line}, ::Type{QuadraticLine}) = ((:here,1), (:here,2), (:there,1), (:there,2), (:there,3))
+get_sides_and_base_indices(::Type{QuadraticLine}, ::Type{QuadraticLine}) = ((:here,1), (:here,2), (:there,1), (:there,2), (:here,3), (:there,3))
+
+get_sides_and_base_indices(::Type{Triangle}, ::Type{Triangle}) = ((:here,1), (:here,2), (:here,3), (:there,1), (:there,2), (:there,3))
+get_sides_and_base_indices(::Type{QuadraticTriangle}, ::Type{Triangle}) = ((:here,1), (:here,2), (:here,3), (:there,1), (:there,2), (:there,3), (:here,4), (:here,5), (:here,6))
+get_sides_and_base_indices(::Type{Triangle}, ::Type{QuadraticTriangle}) = ((:here,1), (:here,2), (:here,3), (:there,1), (:there,2), (:there,3), (:there,4), (:there,5), (:there,6))
+get_sides_and_base_indices(::Type{QuadraticTriangle}, ::Type{QuadraticTriangle}) = ((:here,1), (:here,2), (:here,3), (:there,1), (:there,2), (:there,3), (:here,4), (:here,5), (:here,6), (:there,4), (:there,5), (:there,6))
+
+get_sides_and_base_indices(::Type{Quadrilateral}, ::Type{Quadrilateral}) = ((:here,1), (:here,2), (:here,3), (:here,4), (:there,1), (:there,2), (:there,3), (:there,4))
+get_sides_and_base_indices(::Type{QuadraticQuadrilateral}, ::Type{Quadrilateral}) = ((:here,1), (:here,2), (:here,3), (:here,4), (:there,1), (:there,2), (:there,3), (:there,4), (:here,5), (:here,6), (:here,7), (:here,8), (:here,9))
+get_sides_and_base_indices(::Type{Quadrilateral}, ::Type{QuadraticQuadrilateral}) = ((:here,1), (:here,2), (:here,3), (:here,4), (:there,1), (:there,2), (:there,3), (:there,4), (:there,5), (:there,6), (:there,7), (:there,8), (:there,9))
+get_sides_and_base_indices(::Type{QuadraticQuadrilateral}, ::Type{QuadraticQuadrilateral}) = ((:here,1), (:here,2), (:here,3), (:here,4), (:there,1), (:there,2), (:there,3), (:there,4), (:here,5), (:here,6), (:here,7), (:here,8), (:there,5), (:there,6), (:there,7), (:there,8), (:here,9), (:there,9))
+
+function get_node_ids(c::InterfaceCell)
+    return  collect( getproperty(c, side).nodes[baseindex] for (side, baseindex) in get_sides_and_base_indices(c) )
```

Comment:
Perhaps this would be non-allocating: (?)
```
side_index = get_side_and_base_indeces(c)
n = length(side_index)
return ntuple( i -> getproperty(c, side_index[1]).nodes[side_index[2]), n)
```

Reply:
Good point! :+1:

Reply:
`function_value_jump` etc. are now implemented.
I also fixed a bug related to evaluating gradients.